elastic/eui
# [EuiSuperDatePicker] Allow the `Absolute` tab to accept standardized ISO 8601, RFC 2822, and Unix timestamps

## Summary

closes https://github.com/elastic/kibana/issues/159098 (will take a week or two to reach Kibana main, however)

This PR uses Moment's [String + Formats](https://momentjs.com/docs/#/parsing/string-formats/) API to attempt to parse non-`dateFormat` user input with several standardized timestamp formats as a fallback. I recommend [reviewing by commit](https://github.com/elastic/eui/pull/7331/commits) as I added a few misc cleanup items along the way.

<img width="460" alt="" src="https://github.com/elastic/eui/assets/549407/5d86a0e2-f5e1-4634-b334-23dbbbd64b54">

## QA

- Go to https://eui.elastic.co/pr_7331/#/templates/super-date-picker
- Click the first example's `Last 30 minutes` text
- Click the `Absolute` tab
- Paste the following text into the start date input (sourced from https://www.unixtimestamp.com), and confirm they are all accepted and are correctly formatted to the expected output:
    - [x] `2023-10-31T23:12:10+00:00`
    - [x] `2023-10-31T23:12:10Z`
    - [x] `Tue, 31 Oct 2023 23:12:10 +0000`
    - [x] `1698793930`
- [x] Type in gibberish (e.g. `asdf`) or an invalid date (e.g. `Jan 1st`) or an empty string, and confirm the input still flags these as invalid

### General checklist

- Browser QA
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
    ~- [ ] Checked in both **light and dark** modes~
    ~- [ ] Checked in **mobile**~
- Docs site QA - N/A - I skipped documenting this as there isn't really any existing docs in the first place for the input, and I think the error message covers this 🤔 But LMK if you disagree
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist
    ~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
